### ISSUE TITLE
Establish a process to manage the `Application/*` & `AWS/*` Bicep types generation

### ISSUE THREAD

**Issue by asilverman** [radius-project/radius#6410]
## Impact
TL;DR - Requests to create an AWS resource using UCP are not stable because they may fail if invoked using a stale AWS type definition. 

Consider the following scenario:

Alan (infrastructure operations role) has invested several weeks building a Radius recipe that models AWS resources. These recipes define resources using the Bicep format, assume Alan created the recipe below:
```bicep
import aws as aws

resource foo 'AWS.DynamoDB/Table@default' = {...}
```

The Bicep type definitions are a derivative of the AWS Cloud Formation definitions. The Bicep type definitions drift over time as AWS releases updates on the CloudFormation schema of  `AWS.DynamoDB/Table`.

During deployment time, Boris (developer role) references Alains recipe to model a Radius application. When Boris deploys the application manifest  using `rad`, this eventually translates to an AWS CloudControl API call to create the AWS resource.

The problem is that CloudControl may return an error when the schema of the resource is adrift from the 'live-head' which results in a deployment failure for Boris that is rooted in the recipe Alain published.

Effectively the label @default is the culprit because UCP can't tell if the type version being used  for `AWS.DynamoDB/Table` is conforming with the 'live-head'

This results in unstable application models since there is no guarantee that a previously working model will deterministically deploy successfully. It may fail when:

1. AWS makes a breaking change on the resource type schema
1. The Bicep compiler definitions for AWS are out of date with respect to the 'live-head' published by AWS

## Background

Radius uses Bicep Extensibility Providers (BEPs) to express application models in the Bicep language. Providers have two components: 
 (1) client-side and (2) server-side described in detail in [Extensibility Provider Contract](https://github.com/Azure/bicep/issues/3565). 

The client-side component of the BEP is used by the Bicep compiler and Bicep IDE extensions to validate expressions against the schema published by the BEP to detect compile-time errors and offer IntelliSense suggestions (like auto-completion of required fields and type validation of resource properties)

Example 1: BEP client-side used by Bicep VSCode Extension to detect compile-time errors
![image](https://user-images.githubusercontent.com/9611108/215880767-1b9b72f3-3ad0-449d-affa-b7c05a06f8ba.png)

Example 2: BEP client-side used by Bicep VSCode Extension to auto-fill required properties
![image](https://user-images.githubusercontent.com/9611108/215880296-35481a7d-bf90-4879-9b7c-c9a2f6b2f950.png)
![image](https://user-images.githubusercontent.com/9611108/215880375-154c821c-28d0-4681-9b11-75568d810283.png)

The server-side component of the BEP is the management-control-plane for the resource types managed by the provider. In the case of Radius this logic is handled by the Radius Universal Control Plane (UCP). UCP implements an HTTP server API that is invoked by the Radius Deployment Engine to perform CRUDL operations on radius/aws resources.

<u>**Type definition versions that are processed by the different layers of Radius need to be explicit, unambiguous and matching for both the client-side and server-side to ensure deterministic processing of a given deployment**</u>

The Radius project maintains two BEPs: 

1. Radius - The collection of all bicep extensibility types with a prefix `Application.XXX/XXX` (For example: `Applications.Core/applications@2022-03-15-privatepreview` and `Applications.Link/mongoDatabases@2022-03-15-privatepreview`) imported into the scope of a bicep file using the `import radius as radius` expression.

1. AWS - The collection of all bicep extensibility types with a prefix `AWS.XXX/XXX` (For example: `AWS.EC2/VPC@default` and `AWS.SQS/Queue@default`) imported into the scope of a bicep file using the `import aws as aws` expression.

Bicep extensibility providers expose their type definitions to the Bicep compiler by through a (internal ARM) serialization format known as `types.json` used by the Bicep compiler and IDE extensions to enable the features in Example 1 and 2 (above). This info is also used to compile Bicep files into the ARM-JSON file sent to the Radius Deployment Engine for processing.

## Problem(s) Description

1. There is no process to update the types versions for Radius and AWS types between releases of project radius, this causes confusion since types with the same version (`default` and `2022-03-15-privatepreview`) mutate between Radius releases.

1. The type definition source for AWS reflects the state of the AWS types at the time they were downloaded, these types are updated by AWS in their own timeline and their versions are not currently reflected in the type definitions (see how the value used is `default`)

1. The workflow that updates the type definitions for Radius types is highly manual and prone to failure.

1. The AWS type definitions change over time and unfortunately the AWS Cloud Control documentation doesn't define on what versions of the Cloud Formation Type Schema it operates on. I assume its always the latest which raises the concern that an operation on a type schema that is outdated may fail unexpectedly

[AB#5928](https://dev.azure.com/azure-octo/e61041b4-555f-47ae-95b2-4f8ab480ea57/_workitems/edit/5928)

**Comment by nicolejms** [comment]
very old so closing
